features/admin.identity-providers.v1/api/fido-trusted-apps.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AsgardeoSPAClient, HttpClientInstance } from "@asgardeo/auth-react";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { HttpMethods } from "@wso2is/core/models";
+import { AxiosError, AxiosResponse } from "axios";
+import { store } from "../../admin.core.v1";
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "../../admin.core.v1/hooks/use-request";
+import { IdentityProviderManagementConstants } from "../constants";
+import { FIDOTrustedAppsResponseInterface } from "../models";
+
+/**
+ * Get an axios instance.
+ */
+const httpClient: HttpClientInstance =
+    AsgardeoSPAClient.getInstance().httpRequest.bind(AsgardeoSPAClient.getInstance());
+
+/**
+ * Hook to get the FIDO trusted apps.
+ *
+ * @param shouldFetch - Should fetch the data.
+ *
+ * @returns FIDO trusted apps list.
+ */
+export const useFIDOTrustedApps = <Data = FIDOTrustedAppsResponseInterface, Error = RequestErrorInterface>(
```

Comment:
Lets break these two functions into their own modules that exports default exports.

- use-fido-trusted-apps.ts
- update-fido-trusted-apps.ts

Reply:
@DilshanSenarath Lets send a followup PR.